tests/testnet-calls.py

```diff
@@ -0,0 +1,343 @@
+
+    # Convert the hash to the Ethereum specific signature format
+    eth_signed_message = encode_defunct(hexstr=message_hash.hex())
+
+    # Sign the message with the private key
+    signature = allocation_account.sign_message(eth_signed_message)
+
+    print("Broadcasting allocation")
+    # create random bytes32 for metadata and subgraphDeploymentID
+    arbitraryBytes32 = os.urandom(32)
+    allocate_txn = mockStaking.functions.allocate(
+        arbitraryBytes32,
+        1000,
+        allocation_account.address,
+        arbitraryBytes32,
+        signature.signature.hex()
+    ).build_transaction({
+        "chainId": CHAIN_ID,
+        "gas": 3550000,  # you might want to adjust this value based on the real gas consumption
+        "gasPrice": w3.to_wei('20', 'gwei'),
+        "nonce": w3.eth.get_transaction_count(indexer_account.address)
+    })
+
+    # Send escrow deposit
+    print("txn hash create allocation: ", send_signed_transaction(indexer_account, allocate_txn).hex())
+
+def send_signed_transaction(account, transaction):
+    """
+    Sign a transaction using the given account and send it.
+    Returns the transaction hash.
+    """
+    # Sign the transaction
+    signed_txn = account.signTransaction(transaction)
+
+    # Send the transaction
+    return w3.eth.send_raw_transaction(signed_txn.rawTransaction)
+
+def deposit_to_escrow(sender_account, receiver, amount):
+    # Prepare the transaction for ERC20 approval
+    approve_txn = erc20.functions.approve(ESCROW_ADDRESS, amount).build_transaction({
+        "chainId": CHAIN_ID,
+        "gas": 550000,   # you might want to adjust this value based on the real gas consumption
+        "gasPrice": w3.to_wei('20', 'gwei'),
+        "nonce": w3.eth.get_transaction_count(sender_account.address)
+    })
+
+    # Send ERC20 approval
+    print("txn hash aprove: ", send_signed_transaction(sender_account, approve_txn).hex())
+    
+
+    deposit_txn = escrow.functions.deposit(receiver, amount).build_transaction({
+        "chainId": CHAIN_ID,
+        "gas": 550000,  # you might want to adjust this value based on the real gas consumption
+        "gasPrice": w3.to_wei('20', 'gwei'),
+        "nonce": w3.eth.get_transaction_count(sender_account.address)
+    })
+
+    try:
+    # Send escrow deposit
+        deposit_txn = send_signed_transaction(sender_account, deposit_txn).hex()
+    except ValueError as e:
+        error = str(e)
+        if "transaction underpriced" in error:
+            deposit_txn = escrow.functions.deposit(receiver, amount).build_transaction({
+                "chainId": CHAIN_ID,
+                "gas": 550000,  # you might want to adjust this value based on the real gas consumption
+                "gasPrice": w3.to_wei('20', 'gwei'),
+                "nonce": w3.eth.get_transaction_count(sender_account.address) + 1
+            })
+            deposit_txn = send_signed_transaction(sender_account, deposit_txn).hex()
```

Comment:
Is this just retrying the exact same thing (except nonce increase) a second time?